fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -597,24 +601,35 @@ public boolean visit(java.lang.String elementName, int elementIndex, Reference r
         if (!REFERENCE.equals(searchParamType)) {
             throw invalidComboException(searchParamType, reference);
         }
-        if (reference.getReference() != null) {
-            ReferenceParmVal p = new ReferenceParmVal();
-            p.setName(searchParamCode);
-            p.setValueString(reference.getReference().getValue());
-            result.add(p);
-        }
 
-        // Make sure we process the identifier if there is one.
-        Identifier identifier = reference.getIdentifier();
-        if (reference.getIdentifier() != null) {
-            TokenParmVal p = new TokenParmVal();
-            p.setName(searchParamCode);
-            if (identifier.getSystem() != null) {
-                p.setValueSystem(identifier.getSystem().getValue());
+        // TODO pass in the bundle if we want to support "a relative URL, which is relative to
+        // the Service Base URL, or, if processing a resource from a bundle, which is relative
+        // to the base URL implied by the Bundle.entry.fullUrl (see Resolving References in Bundles)"
+        try {
+            final String baseUrl = ReferenceUtil.getBaseUrl(null);
```

Comment:
OK - so you moved the reference logic into. the ReferenceUtil

Reply:
Correct, and ReferenceUtil is a much better implementation (logic-wise).